include/mbedtls/check_config.h

```diff
@@ -448,6 +448,18 @@
 #error "MBEDTLS_PLATFORM_TIME_MACRO defined, but not all prerequisites"
 #endif
 
+#if defined(MBEDTLS_PLATFORM_MS_TIME_TYPE_MACRO) &&\
+    ( !defined(MBEDTLS_PLATFORM_C) ||\
+        !defined(MBEDTLS_HAVE_TIME) )
```

Comment:
(minor) if you put both the `||` conditions on the same line it's easier to read:

```
#if defined(MBEDTLS_PLATFORM_MS_TIME_TYPE_MACRO) && \
    (!defined(MBEDTLS_PLATFORM_C) || !defined(MBEDTLS_HAVE_TIME))
#error "MBEDTLS_PLATFORM_MS_TIME_TYPE_MACRO defined, but not all prerequisites"
#endif
```